[abseil] build failure

**Host Environment**

- OS: windows10 desktop
- Compiler: arm-uwp

**To Reproduce**

Steps to reproduce the behavior:
`./vcpkg install --recurse abseil:arm-uwp`

**Failure logs** 
|item|message|
|-|-|
|file| [absl/time/internal/cctz/src/zone_info_source.cc](https://github.com/abseil/abseil-cpp/blob/master/absl/time/internal/cctz/src/zone_info_source.cc)
|error| fatal error C1189: #error:  Unsupported MSVC platform

```
#if defined(_M_IX86)
#elif defined(_M_IA_64) || defined(_M_AMD64) || defined(_M_ARM64)
#else
#error Unsupported MSVC platform
#endif  // _M_<PLATFORM>
```

No code here to capture **_M_ARM**

**Additional context**

I had try to modify with a [patch to capture **_M_ARM**](https://github.com/suhao/pkg/commit/c1cd17dd2b9982e9686b73236e8b587f7239808f) which just to confirm that it can be compiled through.  I am not sure if this condition is handled correctly. 

Hope that friends who are good at ARM-UWP could give a good solution to solve this error.  Thank you!

```
From 5968dc74a7b3c326387db1d985e0f5eb0c3d273f Mon Sep 17 00:00:00 2001
Subject: [PATCH 12/12] ARM-UWP
Url: https://github.com/suhao/pkg/commit/c1cd17dd2b9982e9686b73236e8b587f7239808f
---
 absl/time/internal/cctz/src/zone_info_source.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/absl/time/internal/cctz/src/zone_info_source.cc b/absl/time/internal/cctz/src/zone_info_source.cc
index 98ea161..a155fe8 100644
--- a/absl/time/internal/cctz/src/zone_info_source.cc
+++ b/absl/time/internal/cctz/src/zone_info_source.cc
@@ -83,7 +83,7 @@ ZoneInfoSourceFactory default_factory = DefaultFactory;
     "@@U?$default_delete@VZoneInfoSource@cctz@time_internal@" ABSL_INTERNAL_MANGLED_NS                                   \
     "@@@std@@@std@@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@@Z@" ABSL_INTERNAL_MANGLED_BACKREFERENCE \
     "@@ZA")
-#elif defined(_M_IA_64) || defined(_M_AMD64) || defined(_M_ARM64)
+#elif defined(_M_IA_64) || defined(_M_AMD64) || defined(_M_ARM64) || defined(_M_ARM)
 #pragma comment(                                                                                                          \
     linker,                                                                                                               \
     "/alternatename:?zone_info_source_factory@cctz_extension@time_internal@" ABSL_INTERNAL_MANGLED_NS                     \
-- 
2.26.0.windows.1
```

--- a/ports/abseil/CONTROL
+++ b/ports/abseil/CONTROL
@@ -1,5 +1,5 @@
 Source: abseil
-Version: 2020-03-03-4
+Version: 2020-03-03-5
 Homepage: https://github.com/abseil/abseil-cpp
 Description: an open-source collection designed to augment the C++ standard library.
   Abseil is an open-source collection of C++ library code designed to augment the C++ standard library. The Abseil library code is collected from Google's own C++ code base, has been extensively tested and used in production, and is the same code we depend on in our daily coding lives.

--- a/ports/abseil/fix-arm-build.patch
+++ b/ports/abseil/fix-arm-build.patch
@@ -1,0 +1,14 @@
+diff --git a/absl/time/internal/cctz/src/zone_info_source.cc b/absl/time/internal/cctz/src/zone_info_source.cc
+index 98ea161..7209533 100644
+--- a/absl/time/internal/cctz/src/zone_info_source.cc
++++ b/absl/time/internal/cctz/src/zone_info_source.cc
+@@ -83,7 +83,8 @@ ZoneInfoSourceFactory default_factory = DefaultFactory;
+     "@@U?$default_delete@VZoneInfoSource@cctz@time_internal@" ABSL_INTERNAL_MANGLED_NS                                   \
+     "@@@std@@@std@@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@@Z@" ABSL_INTERNAL_MANGLED_BACKREFERENCE \
+     "@@ZA")
+-#elif defined(_M_IA_64) || defined(_M_AMD64) || defined(_M_ARM64)
++#elif defined(_M_IA_64) || defined(_M_AMD64) || defined(_M_ARM) || \
++    defined(_M_ARM64)
+ #pragma comment(                                                                                                          \
+     linker,                                                                                                               \
+     "/alternatename:?zone_info_source_factory@cctz_extension@time_internal@" ABSL_INTERNAL_MANGLED_NS                     \

--- a/ports/abseil/portfile.cmake
+++ b/ports/abseil/portfile.cmake
@@ -5,6 +5,9 @@ set(ABSEIL_PATCHES
 
     # This patch is an upstream commit, the related PR: https://github.com/abseil/abseil-cpp/pull/637
     fix-MSVCbuildfail.patch
+    
+    # Remove this patch in next update, see https://github.com/google/cctz/pull/145
+    fix-arm-build.patch
 )
 
 if("cxx17" IN_LIST FEATURES)

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -44,7 +44,6 @@
 7zip:x64-linux=fail
 7zip:x64-osx=fail
 7zip:x64-uwp=fail
-abseil:arm-uwp=fail
 activemq-cpp:x64-windows-static=fail
 activemq-cpp:x64-linux=fail
 activemq-cpp:x64-osx=fail
@@ -563,7 +562,6 @@ graphite2:arm-uwp=fail
 graphite2:x64-uwp=fail
 graphqlparser:arm-uwp=fail
 graphqlparser:x64-uwp=fail
-grpc:arm-uwp=fail
 gsl:arm-uwp=fail
 gsl:x64-uwp=fail
 # https://github.com/microsoft/vcpkg/pull/11048

